src/file_types.py

```diff
@@ -1,16 +1,18 @@
 import json
+from abc import ABC, abstractmethod
 from pathlib import Path
 
 
-class BaseFileType:
+class BaseFileType(ABC):
```

Comment:
Forces you to implement the marked abstractmethod in a given subclass. The difference being an Attribute error is thrown on subclass instantiation instead of raising a NotImplementedError when calling the missing method post instantiation. It does some other stuff as well, worth reading about